utoipa-gen/src/security_requirement.rs

```diff
@@ -12,23 +12,43 @@ use crate::Array;
 
 #[derive(Default)]
 #[cfg_attr(feature = "debug", derive(Debug))]
-pub struct SecurityRequirementAttr {
-    name: Option<String>,
-    scopes: Option<Vec<String>>,
+pub struct SecurityRequirementsAttrItem {
+    pub name: Option<String>,
+    pub scopes: Option<Vec<String>>,
 }
 
-impl Parse for SecurityRequirementAttr {
+#[derive(Default)]
+#[cfg_attr(feature = "debug", derive(Debug))]
+pub struct SecurityRequirementsAttr(Vec<SecurityRequirementsAttrItem>);
+
+impl Parse for SecurityRequirementsAttr {
     fn parse(input: ParseStream) -> syn::Result<Self> {
+        let mut items = Vec::new();
+
         if input.is_empty() {
-            return Ok(Self {
-                ..Default::default()
-            });
+            return Ok(Self(items));
         }
+
+        items.push(input.parse::<SecurityRequirementsAttrItem>()?);
+
+        while input.lookahead1().peek(Token![,]) {
+            input.parse::<Token![,]>()?;
+            items.push(input.parse::<SecurityRequirementsAttrItem>()?);
+        }
```

Comment:
While this works, you could probably even leverage this: `parse_utils::parse_punctuated_within_parenthesis` or at least change this to `Punctuated::<SecurityRequirementsAttrItem, Comma>::parse_terminated(input)`.